components/kyma-environment-broker/internal/storage/postsql/read.go

```diff
@@ -282,13 +282,15 @@ func (r readSession) GetOperationByTypeAndInstanceID(inID string, opType interna
 func (r readSession) GetOperationsByTypeAndInstanceID(inID string, opType internal.OperationType) ([]dbmodel.OperationDTO, dberr.Error) {
 	idCondition := dbr.Eq("instance_id", inID)
 	typeCondition := dbr.Eq("type", string(opType))
+	stateCondition := dbr.Neq("state", orchestration.Pending)
 	var operations []dbmodel.OperationDTO
 
 	_, err := r.session.
 		Select("*").
 		From(OperationTableName).
 		Where(idCondition).
 		Where(typeCondition).
+		Where(stateCondition).
```

Comment:
Are you sure it's safe in all calls to not return Pending operations?
It might be OK for the runtimes endpoint case, but introduce big issue in others.

Reply:
there will be affect to other calls. it's better to not change the underlying db function..
I changed the logic again to only modify the `ApplySuspensionOperations` and `ApplyUnsuspensionOperations` to not convert pending operation to runtime dto. then mop-agent will not get the wrong state for the below scenario..

> when the runtime is in progress of provisioning but deprovisioning is on hold (state is pending for deprovisioning as the runtime is still in progress of provisioning)